Lib/Logic_layer/LL.c

```diff
@@ -0,0 +1,153 @@
+/**
+  ******************************************************************************
+  * @file    LL.c
+  * @author  Mike
+  * @brief   Controls incomming commands and goes trough all the commands that
+  * are send.
+  ******************************************************************************
+  */
+#include "LL.h"
+/**
+  * @brief  converts text into color.
+  * @param  text color with text see colortext array for reference.
+  * @retval returns a 8-bit color code.
+  */
+uint8_t LL_textToColor(char *text)
+{
+	uint8_t color;
+	int i = 0;
+	char colortext[AMOUNTOFCOLORS][20] = {{"zwart"}, {"blauw"},{"lichtblauw"},
+										 {"groen"}, {"lichtgroen"}, {"cyaan"},
+										 {"lichtcyaan"}, {"rood"}, {"lichtrood"},
+										 {"magenta"}, {"lichtmagenta"}, {"bruin"},
+										 {"geel"}, {"grijs"}, {"wit"}, {"roze"}};
+	for(i=0;i<AMOUNTOFCOLORS;i++){
+		if(strstr(text, colortext[i]) != NULL) //check which color it is
+		{
+			switch(i){ //index of array. assigns a color to the index
+			case 0:
+				color = VGA_COL_BLACK; break;
+			case 1:
+				color = VGA_COL_BLUE; break;
+			case 2:
+				color = VGA_COL_LIGHT_BLUE; break;
+			case 3:
+				color = VGA_COL_GREEN; break;
+			case 4:
+				color = VGA_COL_LIGHT_GREEN; break;
+			case 5:
+				color = VGA_COL_CYAN; break;
+			case 6:
+				color = VGA_COL_LIGHT_CYAN; break;
+			case 7:
+				color = VGA_COL_RED; break;
+			case 8:
+				color = VGA_COL_LIGHT_RED; break;
+			case 9:
+				color = VGA_COL_MAGENTA; break;
+			case 10:
+				color = VGA_COL_LIGHT_MAGENTA; break;
+			case 11:
+				color = VGA_COL_BROWN; break;
+			case 12:
+				color = VGA_COL_YELLOW; break;
+			case 13:
+				color = VGA_COL_GRAY; break;
+			case 14:
+				color = VGA_COL_WHITE; break;
+			case 15:
+				color = VGA_COL_PINK; break;
+			default:
+				return 1;
+			}
+		}
+	}
+	return color;
+}
+
+/**
+  * @brief  Reads command from struct and select which function to run. stops when type = -1
+  * @param  struct command c struct with info of commmand see @ref "file with struct".
+  * @param  last_place from where in the command struct to execute. return value from parser_receiveData
+  * @retval err error with error code @ref LL.h.
+  */
+uint8_t LL_executeCommand(COMMAND *c, uint8_t last_place)
+{
+	uint8_t err = 0;
+	int i = last_place;
+	uint8_t repeat_amount = 0;
+	uint8_t repeat_start = 0;
+	while(c[i].ID != -1 && i<MAX_CMDS){		//set in parser so we know when were at the last message
```

Comment:
spelling: when were -> when we're (ik kan dat soort dingen niet ontzien...)

Reply:
fixed